drivers/bluetooth/hci/Kconfig.esp32

```diff
@@ -7,23 +7,280 @@ config HEAP_MEM_POOL_ADD_SIZE_ESP_BT
 	default 25600 if ESP_BT_HEAP_SYSTEM
 	default 0
 	help
-	  Make sure there is a minimal heap available for BT driver.
+	  Additional heap size reserved for Bluetooth driver usage.
 
 choice ESP_BT_HEAP
-	prompt "Bluetooth adapter heap in use"
+	prompt "Bluetooth heap type"
 	default ESP_BT_HEAP_SYSTEM
 
-	config ESP_BT_HEAP_SYSTEM
-		bool "Bluetooth adapter use the kernel mempool heap (k_malloc)"
+config ESP_BT_HEAP_SYSTEM
+	bool "Kernel memory pool heap (k_malloc)"
 
-endchoice # ESP_BT_HEAP
+endchoice
+
+menu "ESP32 Bluetooth Controller Configuration"
 
 config ESP32_BT_CONTROLLER_STACK_SIZE
 	int "Bluetooth controller stack size"
 	default 4096
 
 config ESP32_BT_CONTROLLER_TASK_PRIO
-	int "Bluetooth controller task priority level"
+	int "Bluetooth controller task priority"
+	default 2
+
+config ESP32_BT_CTRL_BR_EDR_MAX_SYNC_CONN
+	int "BR/EDR synchronous connection limit (SCO/eSCO)"
+	depends on BT_CLASSIC
+	default 0
+	range 0 3
+	help
+	  Maximum number of synchronous connections (SCO/eSCO).
+	  Each connection uses 2 KB of DRAM.
+
+config ESP32_BT_CTRL_BR_EDR_MAX_SYNC_CONN_EFF
+	int
+	default ESP32_BT_CTRL_BR_EDR_MAX_SYNC_CONN if BT_CLASSIC
+	default 0
+
+config ESP32_BT_CTRL_BLE_MAX_CONN
+	int "BLE connection limit"
+	default 3
+	range 1 9
+	help
+	  Maximum number of BLE connections.
+	  Each connection consumes approximately 1 KB of DRAM.
+
+config ESP32_BT_CTRL_BLE_MAX_CONN_EFF
+	int
+	default ESP32_BT_CTRL_BLE_MAX_CONN
+	default 0
+
+config ESP32_BT_CTRL_BR_EDR_MAX_ACL_CONN
+	int "BR/EDR ACL connection limit"
+	depends on BT_CLASSIC
 	default 2
+	range 1 7
+	help
+	  Maximum number of ACL connections.
+	  Each connection uses approximately 1.2 KB of DRAM.
+
+config ESP32_BT_CTRL_BR_EDR_MAX_ACL_CONN_EFF
+	int
+	default ESP32_BT_CTRL_BR_EDR_MAX_ACL_CONN if BT_CLASSIC
+	default 0
+
+config ESP32_BT_BLE_SLEEP_CLOCK_ACCURACY_INDEX_EFF
+	int
+	default 1
+	help
+	  BLE sleep clock accuracy index (default is 151 ppm to 250 ppm).
+
+config ESP32_BT_CTRL_BR_EDR_MIN_ENC_KEY_SZ_DFT
+	int "BR/EDR minimum encryption key size"
+	depends on BT_CLASSIC
+	default 7
+	range 7 16
+	help
+	  Default minimum encryption key size when initiating encryption.
+
+config ESP32_BT_CTRL_BR_EDR_MIN_ENC_KEY_SZ_DFT_EFF
+	int
+	default ESP32_BT_CTRL_BR_EDR_MIN_ENC_KEY_SZ_DFT if BT_CLASSIC
+	default 0
```

Comment:
You have all these `ESP_BT_CTRL_BR_EDR_*` options, but only the first one has `depends on BT_CLASSIC`? Seems like all of them should have that (or rather put them inside an `if BT_CLASSIC` block). Btw, IMO the prefix should be ESP_BT_CTLR_* and not `ESP_BT_CTRL` to be inline with the existing `BT_CTLR_*` options. Otherwise we'll all the time have typos when writing these because of the inconsistency.

Reply:
Sure, I'll check and work this out.

Reply:
@jhedberg, surround necessary entries with `BT_CLASSIC` and prefixed all entries with `ESP32_` to make it common.

Reply:
Anyway, since you've now prefixed everything with ESP32, the breakage is clearly limited to your platforms, so I'll let you decide, but I find it odd that you would want to "get it right" from the beginning. I know the CTLR spelling introduced by @cvinayak is not the most common one, but it's what we're stuck with, and it'll help avoid mistakes if we're consistent with it for all controller Kconfig options.

Reply:
Oh, one other thing: I hope you've gone through all the existing `BT_CTLR_*` options defined under `subsys/bluetooth/controller`, and are not introducing duplicate ones. If an HCI driver is doing `select BT_CTLR` it should use the `BT_CTLR_*` options as applicable, and also add new ones in case they are generic and not limited to one specific vendor or controller.